controller/node_controller.go

```diff
@@ -1027,60 +1031,75 @@ func (nc *NodeController) syncMultipathd(node *longhorn.Node, namespaces []lhtyp
 	node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeMultipathd, longhorn.ConditionStatusTrue, "", "")
 }
 
-func (nc *NodeController) syncNFSClientVersion(kubeNode *corev1.Node, node *longhorn.Node, namespaces []lhtypes.Namespace) {
-	kernelVersion := kubeNode.Status.NodeInfo.KernelVersion
-	nfsClientVersions := []string{"CONFIG_NFS_V4_2", "CONFIG_NFS_V4_1", "CONFIG_NFS_V4"}
+func (nc *NodeController) checkModulesLoaded(kubeNode *corev1.Node, node *longhorn.Node, namespaces []lhtypes.Namespace) {
+	checkingModules := map[string]string{
+		"CONFIG_DM_CRYPT": "dm_crypt",
+	}
 
-	nsexec, err := lhns.NewNamespaceExecutor(lhtypes.ProcessNone, lhtypes.HostProcDirectory, namespaces)
+	notFoundModulesUsingkmod, err := checkModulesLoadedUsingkmod(checkingModules)
 	if err != nil {
-		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeNFSClientInstalled, longhorn.ConditionStatusFalse,
+		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeModulesLoaded, longhorn.ConditionStatusFalse,
 			string(longhorn.NodeConditionReasonNamespaceExecutorErr),
-			fmt.Sprintf("Failed to get namespace executor: %v", err.Error()))
+			fmt.Sprintf("Failed to check kernel modules: %v", err.Error()))
+		return
+	}
+
+	if len(notFoundModulesUsingkmod) == 0 {
+		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeModulesLoaded, longhorn.ConditionStatusTrue, "", "")
 		return
 	}
 
-	kernelConfigPath := "/boot/config-" + kernelVersion
-	args := []string{kernelConfigPath}
-	if _, err := nsexec.Execute(nil, "ls", args, lhtypes.ExecuteDefaultTimeout); err != nil {
+	kernelConfigPath := kernelConfigFilePathPrefix + kubeNode.Status.NodeInfo.KernelVersion
+	configContent, err := lhio.ReadFileContent(kernelConfigPath)
+	if err != nil {
+		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeModulesLoaded, longhorn.ConditionStatusFalse,
+			string(longhorn.NodeConditionReasonKernelConfigIsNotFound),
+			fmt.Sprintf("Unable to get %v content to check if %v are enabled: %v", kernelConfigPath, checkingModules, err.Error()))
```

Comment:
Listing `notFoundModulesUsingkmod` in the message shall be enough?

Reply:
Modified. Thanks